libvast/vast/option_map.hpp

```diff
@@ -0,0 +1,94 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <string>
+#include <string_view>
+
+#include "vast/optional.hpp"
+#include "vast/expected.hpp"
+#include "vast/data.hpp"
+
+#include "vast/detail/steady_map.hpp"
+
+namespace vast {
+
+/// A map for CLI options.
+class option_map {
+public:
+  // -- types ----------------------------------------------------------------
+
+  using key_type = std::string;
+  using mapped_type = data;
+  using value_type = std::pair<key_type, mapped_type>;
```

Comment:
All other type aliases are defined in terms of `map_type::...`. Is there a particular reason to define `key_type`, `mapped_type` and `value_type` manually?

Reply:
```cpp
  using map_type = detail::steady_map<std::string, data>;
  using key_type = map_type::key_type;
  using mapped_type = map_type::mapped_type;
  using value_type = map_type::value_type;
```
I can change it to this style, than all rely on `map_type`

Reply:
👍